service/header/store.go

```diff
@@ -76,6 +64,22 @@ func newStore(ds datastore.Batching) (*store, error) {
 	}, nil
 }
 
+func (s *store) Init(_ context.Context, initial *ExtendedHeader) error {
```

Comment:
I am ok with this comment. The only concern is about consistency with other methods that do not have comments because they implement an interface and the interface itself has comments.

Reply:
IMO all inits should be documented with *how* they're being initialised. It's nice.

Reply:
Ok, 🤷🏻

Reply:
@liamsi, are you ok with those comments being 1:1 duplicates of the interface comments.

Also, https://www.reddit.com/r/golang/comments/2ynbfm/idiomatic_go_should_interface_methods_have/

Reply:
Better than no comments but even better would be implementation specific comments in the implementatiopn.

Reply:
Agree. The comments for Store are out of the scope of this PR, so I will create an issue for this.

Reply:
ack

Reply:
https://github.com/celestiaorg/celestia-node/issues/446